File: 1481-least-number-of-unique-integers-after-k-removals/1481-least-number-of-unique-integers-after-k-removals.py
class Solution:
    def findLeastNumOfUniqueInts(self, arr: List[int], k: int) -> int:
        id_counts = {}
        for id in arr:
            if id in id_counts:
                id_counts[id] += 1
            else:
                id_counts[id] = 1
        
        sorted_counts = sorted(list(id_counts.values()))
        
        removed_count = 0
        while removed_count < k:
            removed_count += sorted_counts.pop(0)
            if removed_count > k:
                return len(sorted_counts) + 1
        
        return len(sorted_counts)

File: 1481-least-number-of-unique-integers-after-k-removals/1481-least-number-of-unique-integers-after-k-removals.js
/**
 * @param {number[]} arr
 * @param {number} k
 * @return {number}
 */
var findLeastNumOfUniqueInts = function(arr, k) {
    const id_counts = {}
    for (let id of arr) {
        id in id_counts ? id_counts[id] += 1 : id_counts[id] = 1;
    }
    
    const sorted = Object.values(id_counts).sort((a, b) => b - a);
    
    let removed = 0;
    while (removed + sorted[sorted.length-1] <= k) {
        removed += sorted.pop();
    }
    
    return sorted.length;
};